docs/rules.md

```diff
@@ -0,0 +1,11 @@
+# Rules in test runners
+
+## Tests must be run in order.
+
+Presuming we're un-skipping all tests, and want to continue to promote a TDD approach, we should ensure that the tests run in order so that the "next" test is the one that fails. Stop as soon as the first test fails so that a user is only presented with one error message.
+
+## The test runner must stop once a test fails.
```

Comment:
While there are certain individual tests with significant runtime (looking at you, `alphametics`), for the most part, individual test cases run imperceptibly fast. Failing fast will not save appreciable runtime expenses.

Simultaneously, in the presence of optional tests which are allowed to fail, we have to keep running tests even after a single failure. Otherwise, the ordering of tests becomes significant: if an optional test fails, but there are core tests after it, they will not be run. 

Therefore, I would prefer to amend this requirement: 

```suggestion
## The test runner must report only 0 or 1 test failures
```

All other failures should be filtered by the runner.

Reply:
I've checked the .NET test runner, and I don't think I'm able to have it stop when there is an error. That said, as coriolinus has said, we could of course restrict the test runner to only ever report the first test failure (if there is one).